public/app/features/dashboard/state/PanelQueryState.ts

```diff
@@ -274,7 +274,7 @@ export class PanelQueryState {
 
     return {
       state: done ? LoadingState.Done : LoadingState.Streaming,
-      series, // Union of series from response and all streams
+      series: this.sendFrames ? getProcessedDataFrames(series) : [],
```

Comment:
Not sure I get this change. series should already a DataFrame so this seems like it is not really needed or we have a bug in the typing and there is implicit conversion somewhere.

Reply:
Sadly it is needed, we can remove it later maybe. But Prometheus in explore is returning legacy formatted data in the streaming event

Reply:
Added some additional typings and comments, won't solve it but at least it should be more visible what is going on and where is the issue.